src/virtual_dom/vlist.rs

```diff
@@ -104,21 +121,60 @@ impl VDiff for VList {
 
         // Process children
         let mut lefts = self.children.iter_mut();
-        let mut rights = rights.drain(..);
-        loop {
-            match (lefts.next(), rights.next()) {
-                (Some(left), Some(right)) => {
-                    previous_sibling = left.apply(parent, previous_sibling.as_ref(), Some(right));
-                }
-                (Some(left), None) => {
-                    previous_sibling = left.apply(parent, previous_sibling.as_ref(), None);
+        let mut rights_nokeys = Vec::with_capacity(rights.len());
+        let mut rights_lookup = HashMap::with_capacity(rights.len());
```

Comment:
This is going to overallocate. How about we first count how many keyed elements `rights` has?